### PR TITLE
VEN-942 | Persist current page after refresh

### PR DESCRIPTION
**Pull request by Nizar-Rahme**
## Description :sparkles:
- Fix the issue that causes the application to go back to the first page after reloading views with backend-paginated tables
- Fix Eslint warnings for e2e tests

### Closes :no_good_woman:
[VEN-942](https://helsinkisolutionoffice.atlassian.net/browse/VEN-942): Refreshing paginated tables go back to the first page

### Automated tests :gear:️
https://github.com/City-of-Helsinki/berth-reservations-admin/blob/5528d381c244ee468838578d79133973716ae227/browser-tests/customerListView.ts#L30

### Manual testing :construction_worker_man:
- Go to one of the backend paginated tables e.g. Applications List
- Go to the second page
- Refresh the page
- The table should stay and render the second page as expected

### CODE REVIEW

File: src/common/table/Table.tsx [hunk 1]
```diff
@@ -262,19 +263,17 @@ const Table = <D extends { id: string }>({
...
+  const currentSortByState = state.sortBy;
+  const didSortByChanged = equal(initialSortByState, currentSortByState);
 
   useEffect(() => {
-    if (initialSortByState?.id === currentSortByState?.id && initialSortByState?.desc === currentSortByState?.desc)
-      return;
-
-    onSortedColChange?.(currentSortByState);
-  }, [currentSortByState, onSortedColChange, initialSortByState]);
+    onSortedColChange?.(currentSortByState[0]);
```
Comment (ahie) [inline src/common/table/Table.tsx:271]: Should this check if `didSortByChanged === true` before calling `onSortedColChange`?

File: src/common/table/Table.tsx [hunk 2]
```diff
@@ -262,19 +263,17 @@ const Table = <D extends { id: string }>({
...
+  }, [currentSortByState, onSortedColChange, didSortByChanged]);
 
   useEffect(() => {
-    onSortedColsChange?.(state.sortBy);
-  }, [state.sortBy, onSortedColsChange]);
+    onSortedColsChange?.(currentSortByState);
```
Comment (ahie) [inline src/common/table/Table.tsx:275]: Same here.

File: src/common/table/Table.tsx (hunk 1 above)
Comment (Nizar-Rahme) [reply src/common/table/Table.tsx]: Added

File: src/common/table/Table.tsx (hunk 2 above)
Comment (Nizar-Rahme) [reply src/common/table/Table.tsx]: Added

**Review by ahie — APPROVED** [review]
LGTM 👍

**Pull request merged**